event.go

```diff
@@ -49,8 +50,8 @@ type ContextIdentity struct {
 }
 
 // NewEvent initalises and populates a new ApiEvent with
-// event details from a http.Request
-func NewEvent(req *http.Request) (*Event, error) {
```

Comment:
Instead of passing it to the NewEvent function (doesn't seem clean), you could try something like this on the request body: 

```go
if utf8.Valid(req.Body) {
   eventBody = string(r.body)
} else {
  eventBody = base64.StdEncoding.EncodeToString(req.Body)
  isBase64 = true
}
```

Reply:
Does this mean encoding any binary request body even if there's no Binary Media Types?
I'm ok with this behavior though it's not precise.

Reply:
You are right, it should match both checks. It is a binary media type && !utf8.Valid(). If it is a binary payload and the content type is not marked as binary in the list, we let the Lambda function fail, just like it would in the service

Reply:
OK. I added `!utf8.Valid` check.

And I also moved NewEvent call from InvokeHTTP in runtime.go to WrappedHandler in mount.go.
I think it's cleaner than passing isBase64Encoded value around.
Please review this change.
Thanks!